src/main/scala/io/iohk/ethereum/domain/Receipt.scala

```diff
@@ -35,24 +62,33 @@ object Receipt {
   *
   * More description: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-658.md
   */
-case class Receipt(
+case class LegacyReceipt(
```

Comment:
I'm not really sure about the name because it is still used in the new receipts while "legacy" sounds like it is only there as a necessary evil and should not really be used anymore. On the other hand I guess it makes it clear that it corresponds to a `LegacyTransaction`. 

It is up for discussion, but we could do it like this: 

```scala
  case class ReceiptBody(
    postTransactionStateHash: TransactionOutcome, 
    cumulativeGasUsed: BigInt, 
    logsBloomFilter: ByteString, 
    logs: Seq[TxLogEntry]
  )
  
  sealed trait TransactionType
  case object Legacy extends TransactionType
  case object Type01 extends TransactionType

  case class Receipt(transactionType: TransactionType, body: ReceiptBody)
```

What do you think ?

Reply:
I was reluctant to normalize Receipt as being always the same for every transactions, current and coming.

But even without enforcing / implying that every receipts will end up with the current form, we can do something like:
- rename LegacyReceipt -> ReceiptBody
- rename TypedLegacyReceipt -> TypedReceipt (with byte changed to enum, the same one used for Transaction, see next comment)
- keep Type01Receipt extends TypedReceipt(ReceiptBody(...))